tools/clang_tidy/lib/src/options.dart

```diff
@@ -70,10 +71,21 @@ class Options {
     required List<io.File> shardCommandsPaths,
     int? shardId,
   }) {
+    io.File? configPath;
+    if (options.wasParsed('config-file')) {
```

Comment:
Looks like the argument validation is happening further down in `_checkArguments` on line 284.